src/activity_handlers.cpp

```diff
@@ -1838,6 +1838,12 @@ void activity_handlers::reload_finish( player_activity *act, player *p )
     }
 
     std::string msg = _( "You reload the %s." );
+    if( reloadable.get_var( "dirt", 0 ) > 7800 ) {
+        msg =
+            _( "You struggle to reload the fouled %s, but manage to loosen some debris and make it somewhat operational." );
+        g->u.moves -= 2500;
```

Comment:
```suggestion
        p->moves -= 2500;
```
Why drag `g->u` into this?

Reply:
Didn't know p->moves would do the same thing.